### PR TITLE
Fix CI when testing with Symfony 5.1

### PR DESCRIPTION
**Pull request by nicolas-grekas**

### CODE REVIEW

**Comment by ostrolucky** [comment]
Can you provide some backstory please? Why are we not asserting now that tags contain `container.no_preload` *only*, but instead trying to hiding this fact?

**Comment by nicolas-grekas** [comment]
`container.no_preload` is added because the only services that reference the tested listener are commands, and commands propagate this tag to their deps.
Testing this is unrelated to the bundle, so it should just be ignored here.

**Comment by greg0ire** [comment]
The CI still fails, I will push a commit that fixes it shortly.

**Comment by nicolas-grekas** [comment]
The fix for deprecations should be borrowed from master

**Comment by greg0ire** [comment]
Oh didn't know, I'll fix this.

**Review by greg0ire — APPROVED** [review]

**Review by SenseException — APPROVED** [review]

**Comment by ostrolucky** [comment]
You ask for my approval, but I am still uncomfortable with this unset thing. We should change assertions to use `$definition->getTags()['doctrine.event_subscriber']` / `$definition->getTags()['doctrine.event_listener']`, instead of coupling these test cases more tightly to Symfony DI by hardcoding their internal tags. In the end these asertions were wrong. These test cases shouldn't care about other tags, just tags we are touching - and adding every other tag in future to unset() later on just doesn't scale.

**Comment by nicolas-grekas** [comment]
@ostrolucky thanks for the suggestion, I updated the PR accordingly. I hope it's OK for everyone else also.

**Review by ostrolucky — APPROVED** [review]

File: Tests/DependencyInjection/AbstractDoctrineExtensionTest.php [hunk 1]
```diff
@@ -655,7 +654,7 @@ public function testAttachEntityListeners() : void
         $methodCalls = $definition->getMethodCalls();
 
         $this->assertDICDefinitionMethodCallCount($definition, 'addEntityListener', [], 6);
-        $this->assertEquals(['doctrine.event_listener' => [ ['event' => 'loadClassMetadata'] ] ], $definition->getTags());
+        $this->assertSame([ ['event' => 'loadClassMetadata'] ], $definition->getTags()['doctrine.event_subscriber']);
```
Comment (ostrolucky) [inline Tests/DependencyInjection/AbstractDoctrineExtensionTest.php:657]:
```suggestion
        $this->assertSame([ ['event' => 'loadClassMetadata'] ], $definition->getTags()['doctrine.event_listener']);
```

**Review by ostrolucky — CHANGES_REQUESTED** [review]

**Review by ostrolucky — APPROVED** [review]
Thanks for bearing with me!

**Pull request merged**